### PR TITLE
Backend/#20/implementing api with db

### PR DESCRIPTION
**Pull request by Mysterise**
#20

### CODE REVIEW

File: server/dbApi.ts [hunk 1]
```diff
@@ -27,32 +26,32 @@ class Database {
     return db
   }
 
-  dbAdd = async doc => {
+  dbAdd = async (termColName: string, doc) => {
```
Comment (Skithy) [inline server/dbApi.ts]: dbAdd was meant to be a generic one
maybe make another function that uses dbAdd instead. Can be more specific, add types
```
interface Class {}
addClass = async (termColName: string, class: Class) => {}
```

File: server/controllers/api.ts [hunk 2]
```diff
@@ -1,13 +1,27 @@
...
   const params: IGetCourseParams = req.params
-  res.send(params.courseId)
+  const course = await db.dbRead(params.termId, params.courseId)
+
+  if (course) {
+    res.send(course)
+  } else {
+    console.error(
+      'dbRead returned null course. params: ' +
+        util.inspect(params, { showHidden: false, depth: null })
```
Comment (Skithy) [inline server/controllers/api.ts]: hmmm interesting, didn't know this existed

Comment (Mysterise) [reply server/controllers/api.ts]: it's very nice!

File: server/dbApi.ts (hunk 1 above)
Comment (Mysterise) [reply server/dbApi.ts]: will discuss in meeting

**Comment by Mysterise** [comment]
Small related backend change is done in #22

**Pull request merged**